### PR TITLE
feat: ts sdk

### PR DESCRIPTION
**Pull request by aabedraba**
- feat: initial sdk version
- feat: publish package, create wrapper and create example

## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [ ] tested locally
- [ ] tested on cluster
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

## Breaking changes

-

## Changes

-

## Fixes

-

### CODE REVIEW

**Review by vsukhin — APPROVED** [review]

File: sdk/src/generated-client.ts [hunk 1]
```diff
@@ -0,0 +1,2976 @@
+/**
```
Comment (rangoo94) [inline sdk/src/generated-client.ts:1]: This generated code shouldn't be a part of GitHub repository, it should be `.gitignore`d.

Instead, you may add it to `"build"` script in NPM.

File: sdk/src/testkube.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
+import * as generatedClient from "./generated-client";
+
+type InitTestkubeOptions = {
+  url: string;
+  apiToken?: string;
+};
+
+export const initTestkube = ({ url, apiToken }: InitTestkubeOptions) => {
+  generatedClient.defaults.baseUrl = url;
+  console.log("Testkube SDK initialized with url: ", url);
```
Comment (rangoo94) [inline sdk/src/testkube.ts:10]:
```suggestion
```

File: sdk/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,110 @@
+{
```
Comment (rangoo94) [inline sdk/tsconfig.json:1]: Could be cleaner to delete all comments there.

File: sdk/package.json [hunk 4]
```diff
@@ -0,0 +1,24 @@
+{
+  "name": "@testkube/sdk",
+  "version": "0.1.3",
+  "description": "Typescript SDK for running Testkube tests and creating pipelines.",
+  "main": "dist/testkube.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
```
Comment (rangoo94) [inline sdk/package.json:8]: There should be `build` script to build with `tsc` and generate client with `oazapfts`. Additionally, `prepublish` script should call it, so `npm publish` would build it correctly.

```suggestion
  "scripts": {
    "build": "oazapfts ../api/v1/testkube.yaml ./src/generated-client.ts && tsc",
    "prepublish": "npm run build",
    "test": "echo \"Error: no test specified\" && exit 1"
  },
```

File: sdk/package.json [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  "keywords": [],
+  "author": "Abdallah Abedraba",
+  "license": "ISC",
+  "dependencies": {
+    "oazapfts": "^4.7.1"
+  },
+  "devDependencies": {},
```
Comment (rangoo94) [inline sdk/package.json:15]: `typescript` missing

File: sdk/src/testkube.ts [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+};
+
+export const test = (name: string) => {
```
Comment (rangoo94) [inline sdk/src/testkube.ts:13]: Is that actually used?

File: sdk/.gitignore [hunk 7]
```diff
@@ -0,0 +1,2 @@
+dist
+node_modules
```
Comment (rangoo94) [inline sdk/.gitignore:2]: Missing empty line at the end, and `src/generated-client.ts`

File: sdk/example/src/index.ts [hunk 8]
```diff
@@ -0,0 +1,17 @@
+import * as testkube from "@testkube/sdk";
+
+testkube.initTestkube({
+  url: "https://demo.testkube.io/results/v1",
+});
+
+const main = async () => {
```
Comment (rangoo94) [inline sdk/example/src/index.ts:7]: Probably it's better to add information in the `README.md` file how to use it, instead of the example package.

**Review by rangoo94 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**